applications/Unity.GrantManager/modules/Unity.SharedKernel/Utils/LetterPairGenerator.cs

```diff
@@ -0,0 +1,52 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace Unity.Modules.Shared.Utils
+{
+    public partial class LetterPairGenerator
+    {
+        // Protected constructor to prevent direct instantiation
+        protected LetterPairGenerator() { }
+
+        private const int timeoutSeconds = 30; // Set the timeout (1 second for this example)
+
+        // Generate letter pairs from a word
+        private static IEnumerable<string> LetterPairs(string str)
+        {
+            int numPairs = str.Length - 1;
+            for (int i = 0; i < numPairs; i++)
+            {
+                yield return str.Substring(i, 2);  // Generate pair of characters
+            }
+        }
+
+        // Async method to split the string with a timeout using a CancellationToken
+        private static string[] SplitWithTimeout(string str, int timeoutSeconds)
+        {
+
+            // Start a task to perform the Regex.Split asynchronously
+#pragma warning disable SYSLIB1045
+            var task = Regex.Split(str, @"\s+", RegexOptions.None, TimeSpan.FromSeconds(timeoutSeconds));
```

Comment:
I think you may have changed something here at some point - Visual Studio believes that the pragma is not required. - also there is nothing from the outside of this method call to indicate that a task is being returned - no async , cancellation etc..

Reply:
I think sonar qube had issues with the @"\s+" not being in a [RegularExpression(@"\s+""
Annotation. I added the Annotation but then in the SharedKernal it did not have the Nuget package.
I installed the Nuget package for <PackageReference Include="System.Text.RegularExpressions" Version="4.3.1" />
into the Shared Kernel proj but it then complained:
Error (active)	CS0246	The type or namespace name 'GeneratedRegexAttribute' could not be found (

I will take out the pragma now and see if it complains.
One of the issues was around not having a timeout which I fixed, that was where all the awaits came in as I originally had pushed it into a Run with a cancel token timeout, the Timespan Regex timeout is cleaner though